Mathlib/Algebra/Module/Submodule/Pointwise.lean

```diff
@@ -281,4 +282,115 @@ scoped[Pointwise] attribute [instance] Submodule.pointwiseMulActionWithZero
 
 end
 
+section
+
+variable [Semiring R] [AddCommMonoid M] [Module R M]
+
+/--
+let `S ⊆ R` be a set and `N ≤ M` be a submodule, then `S • N` is the smallest submodule containing
+all `r • n` where `r ∈ S` and `n ∈ N`.
+-/
+protected def pointwiseSetSMulSubmodule : HSMul (Set R) (Submodule R M) (Submodule R M) where
+  hSMul s N := sInf { p | ∀ ⦃r : R⦄ ⦃n : M⦄, r ∈ s → n ∈ N → r • n ∈ p }
+
+scoped[Pointwise] attribute [instance] Submodule.pointwiseSetSMulSubmodule
+
+variable (s : Set R) (N : Submodule R M)
+
+lemma mem_set_smul_submodule_def (x : M) :
+  x ∈ s • N ↔
+  x ∈ sInf { p : Submodule R M | ∀ ⦃r : R⦄ {n : M}, r ∈ s → n ∈ N → r • n ∈ p } := Iff.rfl
+
+lemma set_smul_submodule_le (p : Submodule R M)
+  (closed_under_smul : ∀ ⦃r : R⦄ ⦃n : M⦄, r ∈ s → n ∈ N → r • n ∈ p) :
+  s • N ≤ p :=
+sInf_le closed_under_smul
+
+lemma set_smul_submodule_eq_of_le (p : Submodule R M)
+  (closed_under_smul : ∀ ⦃r : R⦄ ⦃n : M⦄, r ∈ s → n ∈ N → r • n ∈ p)
+  (le : p ≤ s • N) :
+  s • N = p :=
+le_antisymm (set_smul_submodule_le s N p closed_under_smul) le
```

Comment:
This indentation is the Lean 3 style, not the Lean 4 style, which would indent all but the first line by an extra level.

Reply:
hopefully, indentation now is okay now